### PR TITLE
perf: improve checkpoint size performance [DET-8960]

### PR DESCRIPTION
**Pull request by NicholasBlaskey**
## Description

Checkpoint size was causing noticeable slowdowns. One area I could reproduce ~3 minute time to queries is in cases with a lot of checkpoints with a lot of entries in resources. 

In the previous query we summed up these entries every time on every checkpoint of every trial being updated. 

This change adds a ```size``` column to the checkpoint to only calculate this once, simplifying the queries.

## Test Plan

Integration test should cover it.

As a spot test run an experiment and check the checkpoint size on it and trials.

## Commentary (optional)

Migration takes ~150ms on latest master database

Somewhat contrived but

Pre change benchmark
```
go test -v -tags=integration -bench .  -run="BenchmarkUpdateCheckpointSize"
goos: linux
goarch: amd64
pkg: github.com/determined-ai/determined/master/internal/db
cpu: Intel(R) Core(TM) i7-10510U CPU @ 1.80GHz
BenchmarkUpdateCheckpointSize
BenchmarkUpdateCheckpointSize-8                1        240147623273 ns/op
PASS
ok      github.com/determined-ai/determined/master/internal/db  240.166s
```

Post change benchmark
```
go test -v -tags=integration -bench .  -run="BenchmarkUpdateCheckpointSize"
goos: linux
goarch: amd64
pkg: github.com/determined-ai/determined/master/internal/db
cpu: Intel(R) Core(TM) i7-10510U CPU @ 1.80GHz
BenchmarkUpdateCheckpointSize
BenchmarkUpdateCheckpointSize-8                1        49652737977 ns/op
PASS
ok      github.com/determined-ai/determined/master/internal/db  49.668s
```

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: master/pkg/model/experiment.go [hunk 1]
```diff
@@ -535,18 +537,21 @@ type CheckpointV1 struct {
 	Framework         string     `db:"framework" json:"framework"`
 	Format            string     `db:"format" json:"format"`
 	DeterminedVersion string     `db:"determined_version" json:"determined_version"`
+	Size              int        `db:"size"`
```
Comment (ioga) [inline master/pkg/model/experiment.go:540]: should this one also be int64 explicitly? we aren't consistent in mapping bigint -> int64 in the codebase at all though.

File: master/internal/db/postgres_checkpoints.go [hunk 2]
```diff
@@ -133,59 +144,43 @@ func (db *PgDB) GroupCheckpointUUIDsByExperimentID(checkpoints []uuid.UUID) (
...
+	SUM(size) FILTER (WHERE state != 'DELETED') AS size,
+	COUNT(*) FILTER (WHERE state != 'DELETED') AS count
+	FROM checkpoints_view
+	WHERE trial_id IN (
+		SELECT trial_id FROM checkpoints_view WHERE uuid IN (?)
+	)
+	GROUP BY trial_id
+) sub
+WHERE trials.id = sub.trial_id
+RETURNING true`, bun.In(checkpoints)).Scan(ctx, &res)
```
Comment (ioga) [inline master/internal/db/postgres_checkpoints.go:166]: `res` seems to be unused here and in the next one?
```suggestion
WHERE trials.id = sub.trial_id`, bun.In(checkpoints)).Exec(ctx)
```

File: master/internal/db/postgres_checkpoints.go [hunk 3]
```diff
@@ -133,59 +144,43 @@ func (db *PgDB) GroupCheckpointUUIDsByExperimentID(checkpoints []uuid.UUID) (
...
-		Where("experiments.id = experiment_id").
-		Exec(context.Background())
-
-	return err
+	err = idb.NewRaw(`
+UPDATE experiments SET checkpoint_size=sub.size, checkpoint_count=sub.count FROM (
+	SELECT experiment_id, SUM(checkpoint_size) AS size, SUM(checkpoint_count) as count FROM trials
+	WHERE experiment_id IN (
+		SELECT experiment_id FROM checkpoints_view WHERE uuid IN (?)
+	)
```
Comment (ioga) [inline master/internal/db/postgres_checkpoints.go:176]: wild optimization idea: you probably could be `RETURNING trials.experiment_id` from the previous update, do `Set` on it in go to make it "distinct", and avoid touching `checkpoints_view` 2nd time here.

File: master/internal/db/postgres_checkpoints.go [hunk 4]
```diff
@@ -74,25 +74,36 @@ func (db *PgDB) GetRegisteredCheckpoints(checkpoints []uuid.UUID) (map[uuid.UUID
...
 
-	_, err = db.sql.Exec(`UPDATE checkpoints_v2 c
-    SET state = 'DELETED'
-    WHERE c.uuid IN (SELECT UNNEST($1::uuid[]))`, deleteCheckpoints)
-	if err != nil {
-		return fmt.Errorf("deleting checkpoints from checkpoints_v2: %w", err)
-	}
-	if len(deleteCheckpoints) > 0 {
-		if err := UpdateCheckpointSize(deleteCheckpoints); err != nil {
+	err := Bun().RunInTx(ctx, nil, func(ctx context.Context, tx bun.Tx) error {
```
Comment (ioga) [inline master/internal/db/postgres_checkpoints.go:82]: great work here on tx-ing queries that should've been tx-ed.

**Review by ioga — APPROVED** [review]

File: master/internal/db/postgres_checkpoints.go (hunk 2 above)
Comment (NicholasBlaskey) [reply master/internal/db/postgres_checkpoints.go:166]: I think```NewRaw()``` doesn't have a ```Exec``` method for some reason

Doing ```NewRaw().NewUpdate().Exec``` also does not seem to work and "forgets" the raw query

**Pull request merged**